### ISSUE TITLE
Cannot send empty message through MQTT

### ISSUE THREAD

**Issue by alex007sirois** [Azure/azure-iot-sdk-python#179]
- **OS and version used:** <VERSION>
Raspberry Pi 3 Model B+ 
Raspbian GNU/Linux 9 (stretch)
Linux raspberrypi 4.14.34-v7+ #1110 SMP Mon Apr 16 15:18:51 BST 2018 armv7l GNU/Linux

- **Python runtime used:** <VERSION>
Python 2.7.13

- **SDK version used:** <VERSION>
azure-iothub-device-client==1.4.2
libboost1.62-all

# Description of the issue:
Instantiating an IoTHubClient and then trying to send an empty message gets it stuck in a loop of fail and retry that is based on the retry policy. The error is present both through MQTT and MQTT_WS, but not any other protocol.

# Code sample exhibiting the issue:
```
from iothub_client import IoTHubClient, IoTHubTransportProvider, IoTHubMessage

connection_string = "connection string"
protocol = IoTHubTransportProvider.MQTT
client = IoTHubClient(connection_string, protocol)

def send_confirmation_callback(message, result, user_context): pass

message = IoTHubMessage("")
client.send_event_async(message, send_confirmation_callback, 0)

```
# Console log of the issue:
```
Error: Time:Tue Aug 21 10:51:57 2018 File:/usr/sdk/src/c/iothub_client/src/iothubtransport_mqtt_common.c Func:IoTHubTransport_MQTT_Common_DoWork Line:2904 Failure result from IoTHubMessage_GetData
Error: Time:Tue Aug 21 10:51:57 2018 File:/usr/sdk/src/c/iothub_client/src/iothubtransport_mqtt_common.c Func:IoTHubTransport_MQTT_Common_DoWork Line:2904 Failure result from IoTHubMessage_GetData
Error: Time:Tue Aug 21 10:51:57 2018 File:/usr/sdk/src/c/iothub_client/src/iothubtransport_mqtt_common.c Func:IoTHubTransport_MQTT_Common_DoWork Line:2904 Failure result from IoTHubMessage_GetData
Error: Time:Tue Aug 21 10:51:57 2018 File:/usr/sdk/src/c/iothub_client/src/iothubtransport_mqtt_common.c Func:IoTHubTransport_MQTT_Common_DoWork Line:2904 Failure result from IoTHubMessage_GetData
Error: Time:Tue Aug 21 10:51:57 2018 File:/usr/sdk/src/c/iothub_client/src/iothubtransport_mqtt_common.c Func:IoTHubTransport_MQTT_Common_DoWork Line:2904 Failure result from IoTHubMessage_GetData
Error: Time:Tue Aug 21 10:51:57 2018 File:/usr/sdk/src/c/iothub_client/src/iothubtransport_mqtt_common.c Func:IoTHubTransport_MQTT_Common_DoWork Line:2904 Failure result from IoTHubMessage_GetData
Error: Time:Tue Aug 21 10:51:57 2018 File:/usr/sdk/src/c/iothub_client/src/iothubtransport_mqtt_common.c Func:IoTHubTransport_MQTT_Common_DoWork Line:2904 Failure result from IoTHubMessage_GetData
```

**Comment by pierreca** [comment]
@alex007sirois that sounds like a bug. thanks for reporting it, we'll look into it asap

**Comment by pierreca** [comment]
Fixed with release v1.4.3 (thanks @olivakar !)

**Comment by az-iot-builder-01** [comment]
@alex007sirois, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
